components/raftstore/src/store/util.rs

```diff
@@ -785,6 +800,88 @@ impl PerfContextStatistics {
     }
 }
 
+#[derive(PartialEq, Eq, Debug)]
+pub enum ConfChangeKind {
+    // Only contains one configuration change
+    Simple,
+    // Enter joint state
+    EnterJoint,
+    // Leave joint state
+    LeaveJoint,
+}
+
+impl ConfChangeKind {
+    pub fn confchange_kind(change_num: usize) -> ConfChangeKind {
+        match change_num {
+            0 => ConfChangeKind::LeaveJoint,
+            1 => ConfChangeKind::Simple,
+            _ => ConfChangeKind::EnterJoint,
+        }
+    }
+}
+
+/// Abstracts over ChangePeerV2Request and (legacy) ChangePeerRequest to allow
+/// treating them in a unified manner.
+pub trait ChangePeerI {
```

Comment:
Can trait `ChangePeerI` from raft-proto be reused?

Reply:
Do you mean `ConfChangeI`? I think it can't, because `ConfChangeI` work on existing `ConfChange`, but what `ChangePeerI` mostly do here is to create `ConfChange` based on `ChangePeer` requests.